xattr/lib_build.py

```diff
@@ -0,0 +1,602 @@
+    }
+}
+
+static ssize_t xattr_removexattr(const char *path, const char *name, int options) {
+    if (!(options == 0 || options == XATTR_XATTR_NOFOLLOW)) {
+        return -1;
+    }
+    if (options & XATTR_XATTR_NOFOLLOW) {
+        return lremovexattr(path, name);
+    } else {
+        return removexattr(path, name);
+    }
+}
+
+
+static ssize_t xattr_listxattr(const char *path, char *namebuf, size_t size, int options) {
+    if (!(options == 0 || options == XATTR_XATTR_NOFOLLOW)) {
+        return -1;
+    }
+    if (options & XATTR_XATTR_NOFOLLOW) {
+        return llistxattr(path, namebuf, size);
+    } else {
+        return listxattr(path, namebuf, size);
+    }
+}
+
+static ssize_t xattr_fgetxattr(int fd, const char *name, void *value, ssize_t size, u_int32_t position, int options) {
+    if (position != 0 || !(options == 0 || options == XATTR_XATTR_NOFOLLOW)) {
+        return -1;
+    }
+    if (options & XATTR_XATTR_NOFOLLOW) {
+        return -1;
+    } else {
+        return fgetxattr(fd, name, value, size);
+    }
+}
+
+static ssize_t xattr_fsetxattr(int fd, const char *name, void *value, ssize_t size, u_int32_t position, int options) {
+    int nofollow;
+    if (position != 0) {
+        return -1;
+    }
+    nofollow = options & XATTR_XATTR_NOFOLLOW;
+    options &= ~XATTR_XATTR_NOFOLLOW;
+    if (options == XATTR_XATTR_CREATE) {
+        options = XATTR_CREATE;
+    } else if (options == XATTR_XATTR_REPLACE) {
+        options = XATTR_REPLACE;
+    } else if (options != 0) {
+        return -1;
+    }
+    if (nofollow) {
+        return -1;
+    } else {
+        return fsetxattr(fd, name, value, size, options);
+    }
+}
+
+static ssize_t xattr_fremovexattr(int fd, const char *name, int options) {
+    if (!(options == 0 || options == XATTR_XATTR_NOFOLLOW)) {
+        return -1;
+    }
+    if (options & XATTR_XATTR_NOFOLLOW) {
+        return -1;
+    } else {
+        return fremovexattr(fd, name);
+    }
+}
+
+
+static ssize_t xattr_flistxattr(int fd, char *namebuf, size_t size, int options) {
+    if (!(options == 0 || options == XATTR_XATTR_NOFOLLOW)) {
+        return -1;
+    }
+    if (options & XATTR_XATTR_NOFOLLOW) {
+        return -1;
+    } else {
+        return flistxattr(fd, namebuf, size);
+    }
+}
+
+#else /* Mac OS X assumed */
+#define xattr_getxattr getxattr
+#define xattr_fgetxattr fgetxattr
+#define xattr_removexattr removexattr
+#define xattr_fremovexattr fremovexattr
+#define xattr_setxattr setxattr
+#define xattr_fsetxattr fsetxattr
+#define xattr_listxattr listxattr
+#define xattr_flistxattr flistxattr
+
+/* define these for use in python (see below) */
+#define XATTR_XATTR_NOFOLLOW	XATTR_NOFOLLOW
+#define XATTR_XATTR_CREATE	XATTR_CREATE
+#define XATTR_XATTR_REPLACE	XATTR_REPLACE
+#define XATTR_XATTR_NOSECURITY	XATTR_NOSECURITY
+#endif
+
+#ifndef XATTR_MAXNAMELEN
+#define XATTR_MAXNAMELEN 127
+#endif
+"""
+
+ffi.set_source('_lib', C_SRC)
+
+if __name__ == '__main__':
```

Comment:
By keeping it in there, you can do the lib generation manually which is helpful for debugging. Seems easier than having to run the build script every time. Of course I can remove if you disagree.